### DISCUSSION TITLE
Changing checkout folder structure?

### DISCUSSION THREAD

**Discussion by eduardomps** [actions/checkout#306] [General] [answered]
Hello, I have been tasked with doing some tests in a new project/repo in my org
So far I managed to deploy using a self-hosted runner, using the recommended (PHP) composer workflow which in turn uses **checkout**.

The runner runs in its folder, the workspace is set to the server`s /var/www/_**folder**_ and the checkour is done as /var/www/_**folder/my-repo-name/my-repo-name**_ - it uses the rest api, because of the server's git version being 2.17.

I know there are some path variables available to use in my workflow, but what I couldn`t grasp is: **_can I change the checkout action to checkout the code directly in_** /var/www/folder **_and, if so, how?_**

**Comment by ericsciple** [comment, marked as answer]
Copying the files over might be the best solution currently. I believe rsync is a popular tool for that type of operation. You may want to skip the `.git` folder.

**Reply by eduardomps** [reply]
Thanks, in the end I managed to do just that
I changed the runner to build inside its work folder (the default when installing self-hosted) and, if everything passed, the last step in the build moves it to where it should be

**Reply by macelux** [reply]
@eduardomps I am facing the same problem. Could you share how you changed the runner to build in the /var/www/folder directory ? 
Thanks

**Reply by macelux** [reply]
Never mind , I was able to figure it out using @ericsciple 's suggestion.

**Comment by tanvirstreame** [comment]
Need a good solution
